kernel/src/platform/mpu.rs

```diff
@@ -1,103 +1,109 @@
 //! Interface for configuring the Memory Protection Unit.
 
-#[derive(Debug)]
-pub enum AccessPermission {
-    //                                 Privileged  Unprivileged
-    //                                 Access      Access
-    NoAccess = 0b000,               // --          --
-    PrivilegedOnly = 0b001,         // RW          --
-    UnprivilegedReadOnly = 0b010,   // RW          R-
-    ReadWrite = 0b011,              // RW          RW
-    Reserved = 0b100,               // undef       undef
-    PrivilegedOnlyReadOnly = 0b101, // R-          --
-    ReadOnly = 0b110,               // R-          R-
-    ReadOnlyAlias = 0b111,          // R-          R-
-}
-
-#[derive(Debug)]
-pub enum ExecutePermission {
-    ExecutionPermitted = 0b0,
-    ExecutionNotPermitted = 0b1,
+#[derive(Copy, Clone)]
+pub enum Permission {
+    //                 Privileged  Unprivileged
+    //                 Access      Access
+    NoAccess,       // --          --
+    PrivilegedOnly, // V           --
+    Full,           // V           V
 }
 
+#[derive(Copy, Clone)]
 pub struct Region {
-    base_address: u32,
-    attributes: u32,
+    start: usize,
```

Comment:
Return to `base_address`, and this could probably be a `*const u8`.

Reply:
We'll need to be manipulating these addresses a lot, and use the unsafe offset function every time. Is there any way to do this without using unsafe?